packages/node/src/nodeClient.ts

```diff
@@ -110,6 +102,14 @@ export class NodeClient implements Client<Options> {
       return await Promise.resolve(SKIPPED_RESPONSE);
     }
 
+    if (!isValidEvent(event)) {
+      logger.warn('Found invalid event - skipping logEvent action.');
```

Comment:
Make sense to move validation after middleware. I wonder if we should keep annotate and observer in front so user still have access to that information.

Reply:
I was considering that too but noticed that annotate and observe have previously happened after validation, wasn't sure if that was by design.  @qingzhuozhen do you think it's ok to move them?

Reply:
Looks like `isValidEvent` doesn't reference any of the fields from annotate/observe. I will move them before middleware & validation.

Reply:
I believe in JS these two events don't do much, in others like iOS and android they will populating device information in annotate.